### PR TITLE
Run API and Ingestion Server tests with all relevant ES versions

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Based on https://github.com/WordPress/openverse/pull/2777 and related to the comment here: https://github.com/WordPress/openverse/pull/2777#pullrequestreview-1576169306

## Description

Because production is currently stuck on ES 7 until we've figured out why ES 8 caused the queries to take so long, we need to test all our code against both ES versions. Until we've fully transitioned to ES 8 we need to support both versions.

## Testing Instructions

Check the new CI tasks and confirm they've run as expected. Locally, run `OPENVERSE_ES_VER=7 just api/test` locally to test as an example that the environment variable works as expected.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
This will also need the required checks changed in the infra repo with the new names. I'll push a commit with changes to ingestion and api so that those steps actually run.

**Comment by sarayourfriend** [comment]
The most recent CI run confirms that this runs (and that the changes in #2777 should be okay (I think):

https://github.com/WordPress/openverse/actions/runs/5862236561/job/15893876650?pr=2832#step:5:19

Notice that it pulls ES version 7. And here it pulls version 8:

https://github.com/WordPress/openverse/actions/runs/5862236561/job/15893876557?pr=2832#step:5:19

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -337,6 +350,26 @@ jobs:
...
+      - test-ing
+    strategy:
+      matrix:
+        name:
+          - es_current
+          - es_7
+
+    steps:
+      - name: Pass
+        run: echo 'Ingestion server tests are skipped because ingestion server is unchanged.'
```
Comment (obulat) [inline .github/workflows/ci_cd.yml:371]: Do we really need to update this step since it's not really doing anything? It's just the "Pass" step.

**Review by obulat — APPROVED** [review]
The CI runs prove that this is working.
I've asked inline, too, but do we really need changes to the steps that just "imitate" a CI step, both for the ingestion and the API (the ones that just echo something and don't do anything)?

**Pull request merged**